src/main/java/uk/gov/justice/digital/hmpps/keyworker/events/EventListener.java

```diff
@@ -1,53 +1,109 @@
 package uk.gov.justice.digital.hmpps.keyworker.events;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.jms.annotation.JmsListener;
 import org.springframework.stereotype.Service;
 import uk.gov.justice.digital.hmpps.keyworker.dto.OffenderEvent;
+import uk.gov.justice.digital.hmpps.keyworker.services.KeyworkerService;
 import uk.gov.justice.digital.hmpps.keyworker.services.ReconciliationService;
 
-import java.io.IOException;
-import java.util.Map;
+import java.util.Optional;
 
 @Service
 @ConditionalOnProperty(name = "sqs.provider")
 @Slf4j
+@AllArgsConstructor
 public class EventListener {
-
     private final ObjectMapper objectMapper;
     private final ReconciliationService reconciliationService;
-
-    public EventListener(final ReconciliationService reconciliationService, final ObjectMapper objectMapper) {
-        this.objectMapper = objectMapper;
-        this.reconciliationService = reconciliationService;
-    }
+    private final KeyworkerService keyworkerService;
 
     @JmsListener(destination = "${sqs.queue.name}")
-    public void eventListener(final String requestJson) {
-        final var event = getOffenderEvent(requestJson);
-        if (event != null) {
-            if ("EXTERNAL_MOVEMENT_RECORD-INSERTED".equals(event.getEventType())) {
+    public void eventListener(final String requestJson) throws JsonProcessingException {
+        final var message = getMessage(requestJson);
+        final var eventType = message.getMessageAttributes().getEventType().getValue();
+        log.info("Processing message of type {}", eventType);
+
+        final var event = getOffenderEvent(message.getMessage());
+        switch (eventType) {
+            case "EXTERNAL_MOVEMENT_RECORD-INSERTED":
                 reconciliationService.checkMovementAndDeallocate(event);
-            } else if ("BOOKING_NUMBER-CHANGED".equals(event.getEventType())) {
+                break;
+            case "BOOKING_NUMBER-CHANGED":
                 reconciliationService.checkForMergeAndDeallocate(event);
-            }
+                break;
+            case "DATA_COMPLIANCE_DELETE-OFFENDER":
+                Preconditions.checkState(StringUtils.isNotBlank(event.getOffenderIdDisplay()), String.format("Found null offender id for %s", requestJson));
```

Comment:
meh.. don't need to use String format? takes [template](https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Preconditions.java#L527) anyway?

Reply:
agreed, changed